### ISSUE TITLE
RuntimeError: Grammar refers to 'BeforeKeywordSegment' which was not found in the tsql dialect

### ISSUE THREAD

**Issue by SebbanSms** [sqlfluff/sqlfluff#2956]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Getting an exception when trying to lint a ms sql script

### Expected Behaviour

linting completes without error

### Observed Behaviour

```powershell
WARNING    Unable to lint raw_queries\fake_pdw\sp_slab_l2_attributes.sql due to an internal error. Please report
 this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\linter\runner.py", l
ine 103, in run
    yield partial()
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\linter\linter.py", l
ine 692, in lint_rendered
    parsed = cls.parse_rendered(rendered)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\linter\linter.py", l
ine 369, in parse_rendered
    parsed, pvs = cls._parse_tokens(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\linter\linter.py", l
ine 202, in _parse_tokens
    parsed: Optional[BaseSegment] = parser.parse(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\parser.py", l
ine 36, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 1064, in parse
    self.segments = self.expand(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 438, in expand
    res = stmt.parse(parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 1064, in parse
    self.segments = self.expand(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 438, in expand
    res = stmt.parse(parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 1064, in parse
    self.segments = self.expand(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 438, in expand
    res = stmt.parse(parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\base
.py", line 1000, in parse
    m = parse_grammar.match(segments=segments, parse_context=ctx)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\match_wrapper
.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\ephe
meral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\seque
nce.py", line 134, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\match_wrapper
.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\segments\ephe
meral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\anyof
.py", line 177, in match
    available_options, _ = self._prune_options(
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\anyof
.py", line 75, in _prune_options
    simple = opt.simple(parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\base.
py", line 63, in wrapped_method
    result = func(self, parse_context=parse_context)
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\base.
py", line 783, in simple
    return self._get_elem(dialect=parse_context.dialect).simple(
    File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 804, in _g
et_elem
    return dialect.ref(self._get_ref())
  File "c:\base01\pycharmprojects\cracs\cracs-backend\.venv\lib\site-packages\sqlfluff\core\dialects\base.py", line 232, in ref
    raise RuntimeError(
RuntimeError: Grammar refers to 'BeforeKeywordSegment' which was not found in the tsql dialect

  [1 templating/parsing errors found]

```

### How to reproduce

Not sure how to reproduce this.
The sql query I tried to lint is just a creation of a view.
I cannot add all details here as its not from a database I own.

### Dialect

tsql

### Version

0.11.2

### Configuration

```
[sqlfluff]
# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
# Or run 'sqlfluff dialects'
dialect = tsql

[sqlfluff:rules]
max_line_length = 100
single_table_references = qualified

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = pascal

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper

[sqlfluff:rules:L054]
# GROUP BY/ORDER BY column references
group_by_and_order_by_style = explicit

[sqlfluff:rules:L040]
# Null & Boolean Literals
capitalisation_policy = upper

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
I see you added the SQL but that parses fine for me.

**Comment by SebbanSms** [comment]
> I see you added the SQL but that parses fine for me.

oh, sorry it's the wrong query, let me fix this! I confused the table creation with the view creation (:

**Comment by SebbanSms** [comment]
Updated the query in the issue to the correct one.

**Comment by tunetheweb** [comment]
OK it's the `CREATE TRIGGER` statement.

**Comment by tunetheweb** [comment]
Unfortunately for this type of error it is not possible to ignore this. You will need to exclude the whole file for now (you can add to a `sqfluffignore` file).

I've raised #2957 to add the necessary support.
